Foundation/BUILD

```diff
@@ -0,0 +1,241 @@
+load("@rules_cc//cc:defs.bzl", "objc_library")
+
+exports_files([
+    "GTMLocalizedString.h",
+    "GTMLogger.h"
+])
+
+objc_library(
+    name = "GeometryUtils",
+    hdrs = [
+        "GTMGeometryUtils.h",
+    ],
+    srcs = [
+        "GTMGeometryUtils.m",
+    ],
+    deps = [
+        "//:Defines"
+    ],
+    sdk_frameworks = [
+        "CoreGraphics"
+    ]
+)
+
+objc_library(
+    name = "KVO",
+    hdrs = [
+        "GTMNSObject+KeyValueObserving.h",
+    ],
+    non_arc_srcs = [
+        "GTMNSObject+KeyValueObserving.m",
+    ],
+    deps = [
+        "//:Core",
+        "//DebugUtils:DebugUtils"
+    ],
+    copts = ["-IDebugUtils"],
+    sdk_frameworks = [
+        "CoreGraphics"
+    ]
+)
+
+objc_library(
+    name = "Logger",
+    hdrs = [
+        "GTMLogger.h",
+    ],
+    non_arc_srcs = [
+        "GTMLogger.m",
+    ],
+    deps = [
+        "//:Defines"
+    ]
+)
+
+objc_library(
+    name = "Regex",
+    hdrs = [
+        "GTMRegex.h",
+    ],
+    non_arc_srcs = [
+        "GTMRegex.m",
+    ],
+    deps = [
+        "//:Defines"
+    ],
+    visibility = ["//visibility:public"]
+)
+
+objc_library(
+    name = "StackTrace",
+    hdrs = [
+        "GTMStackTrace.h",
+    ],
+    non_arc_srcs = [
+        "GTMStackTrace.m",
+    ],
+    deps = [
+        "//:Defines"
+    ]
+)
+
+objc_library(
+    name = "StringEncoding",
+    hdrs = [
+        "GTMStringEncoding.h",
+    ],
+    non_arc_srcs = [
+        "GTMStringEncoding.m",
+    ],
+    deps = [
+        "//:Defines"
+    ]
+)
+
+objc_library(
+    name = "SystemVersion",
+    hdrs = [
+        "GTMSystemVersion.h",
+    ],
+    non_arc_srcs = [
+        "GTMSystemVersion.m",
+    ],
+    deps = [
+        "//:Defines"
+    ],
+    visibility = ["//visibility:public"]
+)
+
+objc_library(
+    name = "URLBuilder",
+    hdrs = [
+        "GTMURLBuilder.h"
+    ],
+    non_arc_srcs = [
+        "GTMURLBuilder.m"
+    ],
+    deps = [
+        "//:Core",
+        "//:Defines",
+        ":NSDictionary+URLArguments",
+        ":NSString+URLArguments"
+    ]
+)
+
+objc_library(
+    name = "NSData+zlib",
```

Comment:
NSData_zlib, the `+` won't work with Swift so might as well make it clear the what the module name would be when depended on.